js_utils/axios_helpers.js

```diff
@@ -0,0 +1,25 @@
+"use strict";
+import Vue from "vue";
+import { STATUS } from "@/store/modules/flask/enums";
+
+/**
+ * Handles all HTTP GET calls from Vuex and updates system status if there was an error
+ *
+ * @param {string} whole_url - The entire URL to pass to axios.get
+ * @param {Object} action_context - The context of the Vuex action calling this function (to give this function access to the Vuex store)
+ * @return {Object} the result of the axios call
+ */
+export async function call_axios_get_from_vuex(whole_url, action_context) {
+  let result = 0;
+  try {
+    result = await Vue.axios.get(whole_url);
+  } catch (error) {
+    action_context.commit("flask/set_status_uuid", STATUS.MESSAGE.ERROR, {
+      root: true,
+    });
+    action_context.dispatch("flask/stop_status_pinging"); // Error reported clear the ping_system_status setInterval
```

Comment:
please have tests to confirm this works by checking `state.status_ping_interval_id`.  `flask/stop_status_pinging` is not the name of any action we have, so this line of code currently does nothing (I'm not sure why Vuex doesn't raise an error when attempting to call a function that doesn't exist).

Please have tests from both contexts (`flask` and `playback`) similar to how I tested that setting the system to ERROR worked for both contexts.  This should be able to be tested in the `axios_helpers.spec` test file

Reply:
As per my understanding axios_helpers.js could be axios_common_service.js and its just a common api among all actions. 

So any common API doesn't need a specific .spec file as its just an abstraction of common code, its not have any specific functionality to verify. there is no requirement of one more additional file axios_helpers.spec.js or axios_common_service.js as any way in the spec files of flask/waveform/playback the api is covered via either for 200 or other form of responses is a must requirement to verify.

Reply:
using `context` to commit mutations and dispatch actions behaves differently depending on the module that supplied the context (this is different than then general behavior when using `store.dispatch`).  For example, if you remove line 18 `root: true`, then that action only gets correctly dispatched when this function is called from within the `flask` Vuex module, and does not correctly dispatch when called from `playback`.

Because we are having this interact with Vuex, we need to be a bit more careful about our testing since the `context` changes depending on which module is calling the function